### PR TITLE
User Installation Documentation update

### PR DESCRIPTION
**Pull request by elitest**
See the countershaped output [here](http://secstuff.org/userdoc/install.html)

This PR includes things like:
* de-emphasizing installing pip using the get-pip.py installation method, as Python 2.7.9 ships with pip.
* Linked to developer docs
* I didn't really touch the OS X instructions as I don't have a good way to test those, but if anyone has any comments on them let me know.
* Clarified Ubuntu installation instructions.  Tested on a fresh install of Ubuntu 14.04
* Created Windows installation instructions.  The instructions don't strictly work with the current mitmproxy on PyPI but will when 0.11.4 ships(which I assume is the next time these docs will be updated).  See #554 for more on that.

### CODE REVIEW

**Comment by elitest** [comment]
ref:#398

File: doc-src/install.html [hunk 1]
```diff
@@ -43,17 +38,32 @@
...
+Install the latest version of Python 2 from the <a href="https://www.python.org/downloads/windows/">Python website</a>.  The Windows MSI installer is fairly straightforward and even includes pip as long as you are installing a version of Python 2 after 2.7.9.  For the most part, you can accept the defaults but one option you should change during installation is "Add python.exe to Path".  This will make it easier to run Python.
+<img src="@!urlTo('winpythoninstaller.jpg')!@"/>
 
-- build-essential
-- python-dev
-- libffi-dev
-- libssl-dev
-- libxml2-dev
-- libxslt1-dev
+Now log out of Windows and back in to finish having Python added to your PATH.
```
Comment (mhils) [inline doc-src/install.html]: I think it's sufficient to re-open the terminal nowadays.

File: doc-src/install.html [hunk 2]
```diff
@@ -43,17 +38,32 @@
...
 </pre>
 
+From now on you can run <a href="mitmproxy.html">mitmproxy</a> or <a href="mitmdump.html">mitmdump</a> from a terminal.
+
+## <a id=docUbuntu></a>Installation On Ubuntu
+
+Ubuntu comes with Python but we need to install pip, python-dev and several libraries.  This was tested on a fully patched installation of Ubuntu 14.04.
+
+<pre class="terminal">
+$ sudo apt-get python-pip python-dev libffi-dev libssl-dev libxml2-dev libxslt1-dev
```
Comment (mhils) [inline doc-src/install.html]: + install

File: doc-src/install.html (hunk 1 above)
Comment (elitest) [inline doc-src/install.html]: Terminal?  On Windows?  Not based on my testing, even did a "/taskkill /IM explorer.exe /F".  Couldn't find anything current on the issue, but [this](https://support.microsoft.com/en-us/kb/104011) says:
>However, note that modifications to the environment variables do not result in immediate change. For example, if you start another Command Prompt after making the changes, the environment variables will reflect the previous (not the current) values. The changes do not take effect until you log off and then log back on.

**Comment by mhils** [comment]
:+1:

**Pull request merged**